### ISSUE TITLE
Style linting 

### ISSUE THREAD

**Issue by nijk** [Jarzka/stylefy#22]
I'm not sure if this is going to be possible, but I was wondering if you know of any way to run style linting against the styles provided to or output from this library?

Ideally I'm looking for a solution that would warn developers that they have written styling that will cause issues in the browser. [Stylelint](https://github.com/stylelint/stylelint) provides a good example of the kind of linting that I'm after, but the problem for me is how to get something like this working with Stylefy.

**Comment by Jarzka** [comment]
Outputing stylefy's CSS code into a file would require the browser to write a custom file on your filesystem automatically, which to my understanding, is not possible for security reasons.

**Comment by nijk** [comment]
Ah! I forgot that Stylefy generates the CSS at runtime.

**Comment by nijk** [comment]
AFAICT Garden can be used at compile time - which to my understanding this library uses under the hood. I'm still wondering if there's a way to generate the CSS at compile time for linting purposes. I appreciate there are probably lots of reasons why this might be a bad idea - I'm just wondering if this is even feasible

**Comment by Jarzka** [comment]
The basic problem for compile-time CSS generation is that the **use-style** function is called at runtime, when your component is rendered, and the styles you give it to might have been generated dynamically (for example, if there is some if-else statement which use different style). Thus, I believe creating styles at compile time is difficult and linting is only possible to do via reading the style tags generated by stylefy, if Stylelint supports it.

**Comment by Jarzka** [comment]
I agree that getting that lint feedback at compile time would be useful, but since stylefy works mainly at runtime, this is a bit difficult to achieve. Maybe we could integrate the linting logic of Stylelint into stylefy and run it at runtime too. :D

**Comment by nijk** [comment]
Ok, that's an interesting idea - I imagine that would be a mammoth task though...

I had another mad idea - what about if Stylefy's runtime could send all generated CSS via XHR to a backend that was responsible for linting? That way any linting library could be used and there would be no linting logic to maintain.

**Comment by Jarzka** [comment]
Yes, integration to Stylelint would be huge task...

Interesting idea. I believe it would be possible to take the generated CSS code and send it to somewhere. It would probably be useful in some kind of debugging mode only. :D But since stylefy/init supports option map, it would be easy to use it to turn debugging mode on. :) I think this is an idea worth trying! 

However, I also believe that the bugs you reported are more important to fix right now. :P However, if you wish, feel free to experiment the idea if you want. It should be possible to create such debugging mode without conflicting my bug fixes. But I can also take it to my TODO/TRY list and experiment it when I have time.

**Comment by nijk** [comment]
Oh yes, this is very much a 'nice to' feature request rather than a bug report.

Agreed on the `debugging` mode point, this is a feature that would only be useful in dev and would I imagine come with a very real perf hit.

I'd quite like to understand a little more about the inner workings of this library so I'm happy to have a play around and let you know how I get on.
